index.bs

```diff
@@ -522,6 +522,158 @@ See also:
+</div>
+<div class="example">
+  For geolocation or other physical sensors,
+  no information about what happened while not-"[=Document/fully active=]" should be delivered.
+  The events should simply resume from when the document became "[=Document/fully active=]"
+</div>
+<div class="example">
+  For network connections or streams,
+  the data received while not-"[=Document/fully active=]" should be delivered only
+  when the document becomes "[=Document/fully active=]" again,
+  but whereas a stream might have created many events with a small amount of data each,
+  it could be delivered as smaller number of events with more data in each.
+</div>
+
+<h4 id="omit-non-fully-active">Omit non-"[=Document/fully active=]" documents from APIs that span multiple documents</h4>
+Non-"[=Document/fully active=]" documents should not be observable,
+so APIs should treat them as if they no longer exist.
+They should not be visible to the "outside world" through document-spanning APIs
+(e.g. [clients.matchAll()](https://w3c.github.io/ServiceWorker/#clients-matchall), window.opener).
+
+Note: This should be rare since cross-document-spanning APIs are themselves relatively rare.
+
+<div class="example">
+  [BroadcastChannel](https://html.spec.whatwg.org/multipage/web-messaging.html#broadcasting-to-other-browsing-contexts:fully-active)
+  checks for "[=Document/fully active=]" before sending messages to other browsing contexts.
+</div>
+<div class="example">
+  clients.matchAll()'s [spec](https://w3c.github.io/ServiceWorker/#clients-matchall)
+  currently does not distinguish between "[=Document/fully active=]"
+  and non-"[=Document/fully active=]" clients,
+  but correct implementations should only return "[=Document/fully active=]" clients.
+</div>
+
+<h4 id="discard">Discard non-"[=Document/fully active=]" documents for situations that can't be supported</h4>
+If supporting non-"[=Document/fully active=]" documents is not possible for certain cases,
+explicitly specify it by discarding the document if the situation happens after the user navigated away,
+or setting the document's [salvageable](https://html.spec.whatwg.org/multipage/browsing-the-web.html#concept-document-salvageable)
+bit to false if the situation happens before or during the navigation away from the document,
+to cause it to be automatically discarded after navigation.
+
+Note: this should be rare and probably should only be used when retrofitting old APIs,
+as new APIs should always strive to work well with BFCache.
+
+<div class="example">
+  WebSockets [sets the salvageable bit to false](https://html.spec.whatwg.org/#unloading-documents:concept-document-salvageable-7) during unload.
+</div>
+<div class="example">
+  Calling [clients.claim()](https://w3c.github.io/ServiceWorker/#clients-claim)
+  should not wait for non-"[=Document/fully active=]" clients,
+  instead it should cause the non-"[=Document/fully active=]" client documents to be discarded.
```

Comment:
These "shoulds" are confusing. Does it currently wait for non-fully active clients? I can't find that at a glance. Is there a spec issue about the behavior here?

Reply:
Linked to the issue, do you think we should rewrite it like the matchAll() one above?

Reply:
Either phrasing works; now that you added "this is currently not specified" it's clear.